### PR TITLE
Use Braille to display adjoint and transposed sparse matrices

### LINKED ISSUE #34268: show on sparse adjoints displays dense output
Sparse adjoints display like dense matrices rather than sparse matrices:

```
julia> sprand(10,10,0.5)'
10×10 LinearAlgebra.Adjoint{Float64,SparseMatrixCSC{Float64,Int64}}:
 0.653713   0.0       0.0       0.0       …  0.0        0.132038  0.994473
 0.376673   0.0       0.93067   0.0          0.0        0.889498  0.11722
 0.0        0.0       0.331756  0.0          0.301708   0.68326   0.0
 0.491798   0.0       0.847417  0.0          0.0239453  0.0       0.66232
 0.0        0.457921  0.818687  0.0          0.872932   0.705145  0.42251
 0.0        0.183566  0.0       0.803265  …  0.0        0.929449  0.153552
 0.0        0.0       0.34864   0.0          0.0301167  0.471377  0.277341
 0.0537803  0.183686  0.731216  0.0          0.235513   0.91812   0.0
 0.0        0.845597  0.200048  0.0          0.485561   0.151832  0.768555
 0.0        0.985808  0.756742  0.0          0.595492   0.801145  0.356078
```

### PR DESCRIPTION
**Pull request by goggle**
This PR fixes #34268.
Adjoint and transposed sparse matrices are now printed with Braille, similar to regular sparse matrices.

For some reason, merging the two identical `Base.show` methods into a method
```julia
function Base.show(io::IOContext, S::AbstractSparseMatrixCSCInclAdjointAndTranspose)
```
by using our introduced `const` type, results in weird error, which arises when trying to print a sparse matrix:
```julia
ERROR: MethodError: no method matching display(::SparseMatrixCSC{Float64,Int64})
Closest candidates are:
  display(::Any) at multimedia.jl:322
  display(::AbstractDisplay, ::AbstractString, ::Any) at multimedia.jl:214
  display(::AbstractString, ::Any) at multimedia.jl:215
  ...
Stacktrace:
 [1] display(::Any) at ./multimedia.jl:333
 [2] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [3] invokelatest at ./essentials.jl:709 [inlined]
 [4] print_response(::IO, ::Any, ::Bool, ::Bool, ::Any) at /home/alex/Projects/github.com/julia/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:238
 [5] print_response(::REPL.AbstractREPL, ::Any, ::Bool, ::Bool) at /home/alex/Projects/github.com/julia/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:223
 [6] (::REPL.var"#do_respond#54"{Bool,Bool,Atom.var"#250#251",REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /home/alex/Projects/github.com/julia/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:822
 [7] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [8] invokelatest at ./essentials.jl:709 [inlined]
 [9] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /home/alex/Projects/github.com/julia/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2354
 [10] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /home/alex/Projects/github.com/julia/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:1143
 [11] (::REPL.var"#38#42"{REPL.LineEditREPL,REPL.REPLBackendRef})() at ./task.jl:358
```
It took me so much time to figure that out, and I still don't know why this happened...
So that's the reason why I have that identical (except from the signature) `Base.show` method twice.

### CODE REVIEW

File: stdlib/SparseArrays/src/sparsematrix.jl [hunk 1]
```diff
@@ -226,7 +238,7 @@ end
 Base.show(io::IO, S::AbstractSparseMatrixCSC) = Base.show(convert(IOContext, io), S::AbstractSparseMatrixCSC)
 
 const brailleBlocks = UInt16['⠁', '⠂', '⠄', '⡀', '⠈', '⠐', '⠠', '⢀']
-function _show_with_braille_patterns(io::IOContext, S::AbstractSparseMatrixCSC)
+function _show_with_braille_patterns(io::IOContext, S::AbstractSparseMatrixCSCInclAdjointAndTranspose)
```
Comment (StefanKarpinski) [inline stdlib/SparseArrays/src/sparsematrix.jl:241]: This linear algebra aliases are getting kind of ridiculous. Not a critique, just a humorous observation.

Comment (goggle) [reply stdlib/SparseArrays/src/sparsematrix.jl:241]: Yes, I'm aware of that, I also don't like it.
If anybody has a better solution, feel free to suggest something.

File: stdlib/SparseArrays/src/sparsematrix.jl [hunk 2]
```diff
@@ -298,6 +325,17 @@ function Base.show(io::IOContext, S::AbstractSparseMatrixCSC)
     _show_with_braille_patterns(io, S)
 end
 
+function Base.show(io::IOContext, S::Union{Adjoint{<:Any,<:AbstractSparseMatrixCSC},Transpose{<:Any,<:AbstractSparseMatrixCSC}})
```
Comment (goggle) [inline stdlib/SparseArrays/src/sparsematrix.jl:328]: This is the method which I'm unable to replace with
```
function Base.show(io::IOContext, S::AbstractSparseMatrixCSCInclAdjointAndTranspose)
```
See the error in the description of the PR.

File: stdlib/SparseArrays/src/sparsematrix.jl (hunk 1 above)
Comment (ViralBShah) [reply stdlib/SparseArrays/src/sparsematrix.jl:241]: Not to be tackled in this PR!

**Comment by stevengj** [comment]
Any update on this?

**Comment by goggle** [comment]
I have done a *rebase* to resolve the conflicts.

**Comment by ViralBShah** [comment]
Would be nice to rebase to master and see if we can get this ready for merging.

**Comment by ViralBShah** [comment]
I haven't deeply reviewed the code, but broadly speaking, it looks good to me.

**Comment by simeonschaub** [comment]
Yeah, definitely agree. Will merge later today if there are no more comments.

**Review by ViralBShah — APPROVED** [review]

**Pull request merged**